app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -327,6 +383,10 @@ export default class AMRHeatmapView extends React.Component {
         </p>
       );
     }
+    const [sampleLabels, geneLabels, alleleLabels] = this.extractLabels(
+      samplesWithAMRCounts
+    );
+    const alleleToGeneMap = this.mapAllelesToGenes(samplesWithAMRCounts);
```

Comment:
Since you're calling these methods in the render function (which can potentially get called a lot), it might be good to memoize them. A memoized function will save the last value it returns, and if the next time it's called, the input parameters haven't changed, it will short-circuit and just return the last value again. 

We use `memoize-one` in`CoverageVizBottomSidebar` so you can see how it's used.